front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -879,6 +968,74 @@ export default function AssistantBuilder({
                   timeFrameError={timeFrameError}
                 />
               </div>
+            </div>
+          </div>
+
+          {dustApps.length > 0 && (
+            <div className="flex flex-row items-start">
+              <div className="flex flex-col gap-4">
+                <div className="text-xl font-bold text-element-900">
+                  Dust App Execution
+                </div>
+                <div className="text-sm text-element-700">
+                  Your assistant can execute a Dust App of your design before
+                  answering. The output of the app (last block) is injeced in
+                  context for the model to generate an answer. The inputs of the
+                  app will be automatically infered from the context of the
+                  conversation based on the descriptions you provided in the app
+                  input block dataset.
+                </div>
+                <div className="flex flex-row items-center space-x-2 pt-6">
+                  <div className="text-sm font-semibold text-element-900">
+                    Dust App:
+                  </div>
+                  <DropdownMenu>
+                    <DropdownMenu.Button>
+                      <Button
+                        type="select"
+                        labelVisible={true}
+                        disabled={builderState.dataSourceMode !== "GENERIC"}
```

Comment:
Probably not for now, but it's quite confusing to have the DS become disabled / the dust app become disabled without any explanation

Reply:
cc @Duncid any idea heere?

Reply:
Don't understand what the problem is 😅